.github/workflows/ci.yaml

```diff
@@ -16,19 +16,99 @@ on:
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-java@v2
+    - uses: actions/checkout@v4
+    - uses: actions/setup-java@v3
       with:
-        java-version: '11'
+        java-version: '17'
         distribution: 'temurin'
     - uses: skjolber/maven-cache-github-action@v1
       with:
         step: restore
     - run: mvn -B -U clean verify
+    - uses: actions/upload-artifact@v3
+      with:
+        name: cryostat-core
```

Comment:
We will need to add some unique identifier here. Otherwise, concurrent runs will overwrite each other.